scipy/stats/_hypotests.py

```diff
@@ -389,3 +390,192 @@ def _get_wilcoxon_distr(n):
+      possible combinations of the test statistic, see [2]_.
+
+    The exact calculation will be very slow even for moderate sample
+    sizes as the number of combinations increases rapidly with the
+    size of the samples. If ``method=='auto'``, the exact approach
+    is used if both samples contain less than 10 observations,
+    otherwise the asymptotic distribution is used.
+
+    If the underlying distribution is not continuous, the p-value is likely to
+    be conservative (Section 6.2 in [3]_).
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Cramer-von_Mises_criterion
+    .. [2] Anderson, T.W. (1962). On the distribution of the two-sample
+           Cramer-von-Mises criterion. The Annals of Mathematical
+           Statistics, pp. 1148-1159.
+    .. [3] Conover, W.J., Practical Nonparametric Statistics, 1971.
+
+    Examples
+    --------
+
+    Suppose we wish to test whether two samples generated by
+    ``scipy.stats.norm.rvs`` have the same distribution. We choose a
+    significance level of alpha=0.05.
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> np.random.seed(626)
+    >>> x = stats.norm.rvs(size=100)
+    >>> y = stats.norm.rvs(size=70)
+    >>> res = stats.cramervonmises_2samp(x, y)
+    >>> res.statistic, res.pvalue
+    (0.24331932773109344, 0.19815666195647663)
+
+    The p-value exceeds our chosen significance level, so we do not
+    reject the null hypothesis that the observed samples are drawn from the
+    same distribution.
+
+    For small sample sizes, one can compute the exact p-values:
+
+    >>> x = stats.norm.rvs(size=7)
+    >>> y = stats.t.rvs(df=2, size=6)
+    >>> res = stats.cramervonmises_2samp(x, y, method='exact')
+    >>> res.statistic, res.pvalue
+    (0.2655677655677655, 0.18706293706293706)
+
+    The p-value based on the asymptotic distribution is a good approximation
+    even though the sample size is small.
+    
+    >>> res = stats.cramervonmises_2samp(x, y, method='asymp')
+    >>> res.statistic, res.pvalue
+    (0.2655677655677655, 0.17974247316290415)
+
+    Independent of the method, one would not reject the null hypothesis at the
+    chosen significance level in this example.
+    """
+
+    xa = np.sort(np.asarray(x))
+    ya = np.sort(np.asarray(y))
+
+    if xa.size <= 1 or ya.size <= 1:
+        raise ValueError('x and y must contain at least two observations.')
+    if xa.ndim > 1 or ya.ndim > 1:
+        raise ValueError('The samples must be one-dimensional.')
+    if method not in ['auto', 'exact', 'asymp']:
+        raise ValueError('method must be either auto, exact or asymp.')
+
+    nx = len(xa)
+    ny = len(ya)
+
+    if method == 'auto':
+        if max(nx, ny) > 10:
+            method = 'asymp'
+        else:
+            method = 'exact'
+
+    # get ranks of x and y in the pooled sample
+    z = np.concatenate([xa, ya])
+    # import here to avoid circular import
+    from scipy.stats import rankdata
```

Comment:
Were you running into problems?

Reply:
I remember that I ran into a circular import in this file a while ago but I can double check

Reply:
I think I was running into this sort of thing in gh-13312 and something else recently. Try just `import scipy.stats` at the top and calling `scipy.stats.rankdata` here.

Reply:
thanks, that works. In a follow-up PR, I will change it in a few other places but I would like to keep this PR limited to the CvM 2-sample test

Reply:
Sure.